internal/controller/propagation_controller.go

```diff
@@ -209,6 +217,38 @@ func (r *PropagationReconciler) getVersion(ctx context.Context, propagation v1al
 	return rn.Document().Value, nil
 }
 
+func (r *PropagationReconciler) SetReadyConditionFalse(ctx context.Context, propagation *v1alpha1.Propagation, err error, reason PropagationReadyReason) (ctrl.Result, error) {
+	meta.SetStatusCondition(&propagation.Status.Conditions, metav1.Condition{
+		Type:               v1alpha1.ReadyCondition,
+		Message:            err.Error(),
+		Status:             metav1.ConditionFalse,
+		ObservedGeneration: propagation.Generation,
+		Reason:             string(reason),
+	})
+	r.Client.Status().Update(ctx, propagation)
+	return ctrl.Result{}, err
+}
+
+func deployAfterFromConfig(propagation v1alpha1.Propagation, c config.Config) (*v1alpha1.DeployAfter, error) {
```

Comment:
I have some guesses about what exactly this function is supposed to do based on analyzing what does it do, but would be valuable to have it document the purpose of this function, as this seems to be the core part of overall propagation logic.